site/app/controllers/admin/LateController.php

```diff
@@ -143,43 +144,76 @@ public function updateExtension() {
                 for ($i = 0; $i < count($data); $i++){
                     $this->core->getQueries()->updateExtensions($data[$i][0], $data[$i][1], $data[$i][2]);
                 }
-
-                return $this->getExtensions($data[0][1]);
+                return Response::JsonOnlyResponse(JsonResponse::getSuccessResponse());
             }
         }
         else {
             if ((!isset($_POST['g_id']) || $_POST['g_id'] == "" )) {
                 $error = "Please choose a gradeable_id";
+                $this->core->addErrorMessage($error);
                 return Response::JsonOnlyResponse(
                     JsonResponse::getFailResponse($error)
                 );
             }
             $user = current($this->core->getQueries()->getUsersById([$_POST['user_id']]));
             if (!$user) {
                 $error = "Invalid Student ID";
+                $this->core->addErrorMessage($error);
                 return Response::JsonOnlyResponse(
                     JsonResponse::getFailResponse($error)
                 );
             }
-            if ((!isset($_POST['late_days'])) || $_POST['late_days'] == "" || (!ctype_digit($_POST['late_days']))) {
-                $error = "Late Days must be a nonnegative integer";
+            $late_days = null;
+            if (isset($_POST['late_days']) && $_POST['late_days'] != "") {
+                $late_days = $_POST['late_days'];  
+                if (intval($late_days) < 0 || !ctype_digit($late_days)) {
```

Comment:
This should probably only allow positive integers.  Right now if you give an extension of 0 days it gives a success message then does nothing.

Reply:
We update with a value of 0 to delete extensions, so rather than remove it as an option, I've added an additional handler if no changes are made. It should look something like this: 
![Screenshot from 2019-08-08 14-03-34](https://user-images.githubusercontent.com/32647488/62726506-5cf94480-b9e5-11e9-9a3d-b19b6db84aae.png)